### PR TITLE
[WIP] Fix Markdown table formatting in benchmark output

### PR DESCRIPTION
**Pull request by heynickc**
Per #162 - fixes Markdown table formatting in Benchmark output.  Tagging WIP because a test is yet to be written.

### CODE REVIEW

**Comment by heynickc** [comment]
`continuous-integration/perf-tests` fail because of timeout when running the .NET Core version.  Currently set to 25 minutes, but could increase if needed

**Comment by Aaronontheweb** [comment]
@heynickc does that seem long to you? we don't run THAT many specs...

File: src/NBench/Reporting/Targets/MarkdownBenchmarkOutput.cs [hunk 1]
```diff
@@ -167,7 +168,7 @@ public void WriteBenchmark(BenchmarkFinalResults results)
         private static string BuildStatTable(IEnumerable<AggregateMetrics> metrics, int columnWidth = MaxColumnSize)
         {
             var sb = new StringBuilder();
-            sb.Append(("Metric" + MarkdownTableColumnEnd).PadLeft(columnWidth))
+            sb.Append(MarkdownTableColumnStart + ("Metric" + MarkdownTableColumnEnd).PadLeft(columnWidth))
```
Comment (heynickc) [inline src/NBench/Reporting/Targets/MarkdownBenchmarkOutput.cs]: Adding this extra "|" in the beginning of each row fixes the table formatting in the Markdown benchmark outputs

File: tests/NBench.Tests.End2End/Reporting/MarkdownBenchmarkOutputSpec.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                new BenchmarkResults("NBench.FakeBenchmark",
+                    new BenchmarkSettings(TestMode.Test, RunMode.Iterations, 30, 1000,
+                        new List<IBenchmarkSetting>(),
+                        new ConcurrentDictionary<MetricName, MetricsCollectorSelector>()),
+                    new List<BenchmarkRunReport>()),
+                new List<AssertionResult>());
+            mdOutput.WriteBenchmark(fakeBenchmarkResults);
+            var fakePerfResultsFile = Directory.GetFiles(_perfResultsPath, "NBench.FakeBenchmark*.md",
+                SearchOption.AllDirectories);
+            Approvals.Verify(File.ReadAllText(fakePerfResultsFile.FirstOrDefault()));
```
Comment (heynickc) [inline tests/NBench.Tests.End2End/Reporting/MarkdownBenchmarkOutputSpec.cs]: This test builds a fake `MarkdownBenchmarkOutput` and calls `WriteBenchmark(fakeBenchmarkResults)` to build a Markdown file to be Approved by ApprovalTests.

File: tests/NBench.Tests.End2End/Reporting/MarkdownBenchmarkOutputSpec.cs [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        public MarkdownBenchmarkOutputSpec(ITestOutputHelper ouput)
+        {
+            _output = ouput;
+        }
+
+        [Fact]
+        public void Should_compare_benchmark_output_static_data()
+        {
+            CleanPerfDir(_perfResultsPath);
+            SystemTime.UtcNow = () => new DateTime(2017, 3, 13);
```
Comment (heynickc) [inline tests/NBench.Tests.End2End/Reporting/MarkdownBenchmarkOutputSpec.cs]: Set the time so the header of the benchmark output contains the same time every time it's built.

File: src/NBench/Reporting/Targets/MarkdownBenchmarkOutput.cs [hunk 4]
```diff
@@ -83,7 +84,7 @@ public void WriteBenchmark(BenchmarkFinalResults results)
             {
                 sb.AppendLine($"__{results.Data.Settings.Description}__");
             }
-            sb.AppendLine($"_{DateTime.UtcNow}_");
+            sb.AppendLine($"_{SystemTime.UtcNow()}_");
```
Comment (heynickc) [inline src/NBench/Reporting/Targets/MarkdownBenchmarkOutput.cs]: Switched to use custom `SystemTime.UtcNow` class so the time can be "set" for testing.

Comment (Aaronontheweb) [inline src/NBench/Reporting/Targets/MarkdownBenchmarkOutput.cs]: Virtual time scheduling 4TW

File: src/NBench/Reporting/Targets/MarkdownBenchmarkOutput.cs [hunk 5]
```diff
@@ -284,11 +285,20 @@ private static string BuildRunTable(IReadOnlyList<BenchmarkRunReport> runs, int
...
             sb.AppendLine();
         }
     }
+
+    // used in header of benchmark output markdown file
+    // defaults to DateTime.UtcNow but can be replaced for testing purposes
+    // e.g. NBench.Tests.End2End.Reporting.MarkdownBenchmarkOutputSpec
+    public static class SystemTime
+    {
+        public static Func<DateTime> UtcNow = () => DateTime.UtcNow;
```
Comment (Aaronontheweb) [inline src/NBench/Reporting/Targets/MarkdownBenchmarkOutput.cs]: Make this `internal`

**Pull request merged**